jetty-maven-plugin/pom.xml

```diff
@@ -64,8 +64,12 @@
             <jettyStopKey>${jetty.stopKey}</jettyStopKey>
             <jettyStopPort>${jetty.stopPort}</jettyStopPort>
             <jettyRunPort>${jetty.runPort}</jettyRunPort>
+            <jettyHttpPort>${jetty.httpPort}</jettyHttpPort>
```

Comment:
Nope, this won't work. This is what was there originally that cannot work in this scenario. The problem is that there is significant chance that the port that was found to be free is in fact used by the time the port number is given to the jetty process to start. We shouldn't even really be using this method to obtain stopPort for exactly the same reason.

Reply:
well I'm not sure it is so "significant"..... I used this way of reserving ports for build for long now and never had any issues. 
but maybe there is probably a little chance.
Do you have any other ideas to obtain a random stop port? 
Ok  I don't mind reverting everything I did.
The fix for slow machine is simply maybe to increase the possible wait time to get the forked jetty started (and the file created).

Reply:
The build used to be often broken because the port was grabbed by another process and used before jetty could run. That was before we got the new setup with more hardware, so I guess less chance that the port would be used because the machine should be less loaded than before. Nonetheless, the safest way to do it is to get the process that is going to use the port to be the one to request the port right before use. That means jetty has to get the port.  I think if we can just put a small sleep in after checking for the existence of the port file we should be OK. Alternatively we could change the jetty listener that writes the file to write to a tmp file and then rename to the final location so its atomic: that way when the test checks for the existence of the file the content should be in it.

Reply:
+1 on what Jan says.  I like having a positive selection of the port by the server and a file written to tell others what the port is and that the server is running.   We just need a more robust and patient way of watching for and reading that file.

Reply:
ok democracy majority vote wins :-)